docs/resources/as_planned_task.md

```diff
@@ -0,0 +1,129 @@
+  }
+
+  instance_number {
+    max    = 3
+    min    = 1
+    desire = 2
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `scaling_group_id` - (Required, String, ForceNew) Specifies the AS scaling group ID.
+  Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the name of planned task.
+
+* `scheduled_policy` - (Required, List) Specifies the scheduled or periodic policy for AS planned task.
+  The [scheduled_policy](#AS_ScheduledPolicy) structure is documented below.
+
+* `instance_number` - (Required, List) Specifies the numbers of scaling group instance for AS planned task.
+  The [instance_number](#AS_InstanceNumber) structure is documented below.
+
+<a name="AS_ScheduledPolicy"></a>
+The `scheduled_policy` block supports:
+
+* `launch_time` - (Required, String) Specifies the execution time of the AS planned task.
+  + If `recurrence_type` not set or is empty, the time format is **yyyy-MM-ddTHH:mmZ**.
+  + If `recurrence_type` is specified, the time format is **HH:mm**.
+
+* `recurrence_type` - (Optional, String) Specifies the triggering type of AS planned task.
+  When not set or is empty, the planned task is scheduled execution.
+  After setting, the planned task is periodic execution. The valid values are as follows:
+  + **DAILY**: by day periodic execution.
+  + **WEEKLY**: by week periodic execution.
+  + **MONTHLY**: by month periodic execution.
+
+* `recurrence_value` - (Optional, String) Specifies the frequency at which planned task are triggered.
+  Required only when `recurrence_type` is **WEEKLY** or **MONTHLY**.
+  + When `recurrence_type` is **WEEKLY**, The valid value ranges from `1` to `7`.
+  + When `recurrence_type` is **MONTHLY**, The valid value ranges from `1` to `31`.
+
+* `start_time` - (Optional, String) Specifies the effective start time of the planned task.
+  Only effective when `recurrence_type` is not empty.
+  The time format is **yyyy-MM-ddTHH:mmZ**.
+  If not set, the default is the time when the planned task is successfully created.
+
+* `end_time` - (Optional, String) Specifies the effective end time of the planned task.
+  Only effective and required when `recurrence_type` is not empty.
+  The time format is **yyyy-MM-ddTHH:mmZ**.
+
+<a name="AS_InstanceNumber"></a>
+The `instance_number` block supports:
+
+* `max` - (Optional, Int) Specifies the maximum number of instances for the scaling group. The default value is `0`.
+
+* `min` - (Optional, Int) Specifies the minimum number of instances for the scaling group. The default value is `0`.
+
+* `desire` - (Optional, Int) Specifies the expected number of instances for the scaling group.
+  The default value is `0`.
+
+-> At least set one of `max`, `min` or `desire` parameters.
```

Comment:
According to your code logic, if `min` is set but `desire` is not set, an error will be reported. Also, if `desire` is set but `max` is not set, an error will be reported.